File: js/PicklePeer.js
fan.afPickle.PicklePeer = fan.sys.Obj.$extend(fan.sys.Obj);

fan.afPickle.PicklePeer.prototype.$ctor = function(self) { }

fan.afPickle.PicklePeer.prototype.$typeof = function() {
	return fan.afPickle.PicklePeer.$type;
}

fan.afPickle.PicklePeer.readObjFromIn = function(input, options) {
	if (options === undefined) options = null;
	return new afPickle_ObjDecoder(input, options).readObj();
}

fan.afPickle.PicklePeer.writeObjToOut = function(output, obj, options) {
	if (options === undefined) options = null;
	new afPickle_ObjEncoder(output, options).writeObj(obj);
}

File: java/fan/afPickle/PicklePeer.java
package fan.afPickle;

import fan.sys.Map;
import fan.sys.OutStream;
import fan.sys.InStream;

public class PicklePeer {

	public static Object readObjFromIn(InStream in) { return readObjFromIn(in, null); }
	public static Object readObjFromIn(InStream in, Map options) {
		return new ObjDecoder(in, options).readObj();
	}

	public static void writeObjToOut(OutStream out, Object obj) { writeObjToOut(out, obj, null); }
	public static void writeObjToOut(OutStream out, Object obj, Map options) {
		new ObjEncoder(out, options).writeObj(obj);
	}
}
